packages/react/src/hooks/useIsMountedOnThisRender/useIsMountedOnThisRender.ts

```diff
@@ -0,0 +1,15 @@
+import { useEffect, useRef } from 'react'
+
+export const useIsMountedOnThisRender = () => {
+  const isMountedOnThisRenderRef = useRef(false)
+
+  useEffect(() => {
+    isMountedOnThisRenderRef.current = true
+
+    return () => {
+      isMountedOnThisRenderRef.current = false
+    }
+  }, [])
+
+  return isMountedOnThisRenderRef.current
```

Comment:
I think this value is not changed for using this hook situation.

Because, it`s return value is never change until re-call

for example, I guess like this

```js
let FOO = 1;
const foo = () => FOO;
const bar = () => {
  FOO = 2;
}

const result = foo(); // 1;
bar();
// result is also 1;

const result2 = foo();
// result2 is 2;
```

So, i think it is better return value is not current value, but refObject

```suggestion
  return isMountedOnThisRenderRef
```

plz check my test code result

Reply:
<img width="1310" alt="image" src="https://github.com/suspensive/react/assets/57122180/9f9684cd-bc78-4b00-aa73-f9838c460586">

<img width="1288" alt="image" src="https://github.com/suspensive/react/assets/57122180/3d1e076a-7244-4f6e-a98c-be327895ef0a">

Reply:
As the role of this hook provides has been mounted at this point(calling this hook), why don't you return a function that returns whether it has been mounted?

like this

```ts
export function useIsMounted() {
  const isMountedRef = useRef(true);
  const isMounted = useCallback(() => isMountedRef.current, []);

  useEffect(() => {
    return () => void (isMountedRef.current = false);
  }, []);

  return isMounted;
}
```

[ref](https://stackoverflow.com/questions/58979309/checking-if-a-component-is-unmounted-using-react-hooks)

Reply:
ErrorBoundaryGroup don't need mount checking hook after b1d1a5c. but Thanks for your comment to make me care this hook carefully. I perfectly understood what you want me to how handle react's ref.